### PR TITLE
wallet store: include witness data

### PR DESCRIPTION
**Pull request by sstone**

### CODE REVIEW

File: core/src/wallet.proto [hunk 1]
```diff
@@ -398,9 +403,9 @@ message Wallet {
 
 /** An exchange rate between Bitcoin and some fiat currency. */
 message ExchangeRate {
-  // This much of satoshis (1E-8 fractions)…
+  // This much of satoshis (1E-8 fractions)?
```
Comment (schildbach) [inline core/src/wallet.proto]: This change and the next one is unrelated.

File: core/src/wallet.proto [hunk 2]
```diff
@@ -153,6 +156,8 @@ message TransactionInput {
   optional uint32 sequence = 4;
   // Value of connected output, if known
   optional int64 value = 5;
+  // script witness
+  optional ScriptWitness witness = 10;
```
Comment (schildbach) [inline core/src/wallet.proto]: Any reason for 10? Why not continue with 6?

File: core/src/wallet.proto [hunk 3]
```diff
@@ -22,7 +22,6 @@
  * - Endianness: All byte arrays that represent numbers (such as hashes and private keys) are Big Endian
  * - To regenerate after editing, run: mvn generate-sources -DupdateProtobuf
  */
-
```
Comment (schildbach) [inline core/src/wallet.proto]: Unrelated change.

File: core/src/main/java/org/bitcoinj/core/TransactionWitness.java [hunk 4]
```diff
@@ -64,4 +64,19 @@ public String toString() {
         }
         return Utils.SPACE_JOINER.join(stringPushes);
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        TransactionWitness that = (TransactionWitness) o;
```
Comment (schildbach) [inline core/src/main/java/org/bitcoinj/core/TransactionWitness.java]: Nitpick: We usually use the variable name "other" at that point in equals() impls.

File: core/src/wallet.proto (hunk 3 above)
Comment (sstone) [inline core/src/wallet.proto]: I'll restore the original formatting

File: core/src/wallet.proto (hunk 2 above)
Comment (sstone) [inline core/src/wallet.proto]: Old habit when editing protobuf files, there's no real reason here I'll change it to 6

File: core/src/wallet.proto (hunk 1 above)
Comment (sstone) [inline core/src/wallet.proto]: I'll restore the original formatting

**Comment by schildbach** [comment]
Merged. Thanks a bunch!
